numba/unicode_support.py

```diff
@@ -249,9 +249,18 @@ def _PyUnicode_ToLowercase(ch):
     raise NotImplementedError
 
 
+# From: https://github.com/python/cpython/blob/201c8f79450628241574fba940e08107178dc3a5/Objects/unicodectype.c#L211-L225
 @register_jitable
 def _PyUnicode_ToLowerFull(ch, res):
-    raise NotImplementedError
+    ctype = _PyUnicode_gettyperecord(ch)
+    if (ctype.flags & _PyUnicode_TyperecordMasks.EXTENDED_CASE_MASK):
+        index = ctype.lower & 0xFFFF
+        n = ctype.lower >> 24
+        for i in range(n):
+            res[i] = _PyUnicode_ExtendedCase(index + i)
```

Comment:
Not sure about this, but do you not need to use `_set_code_point` here? The previous functions all seem to do that.

Reply:
`_set_code_point` is implemented in `unicode.py`. So we get import conflict if `unicode_support.py` imports `unicode.py` because `unicode.py` already imports `unicode_support.py`. If to move the code of `_set_code_point` to `unicode_support.py` it will require changes in many different places not related to the feature. Additionally we will need to test changed pieces of code to be sure we didn't break something. So I'm not sure we need to do that in this commit.

Reply:
o.k. no worries, perhaps mark it with a comment and if it passes merge this and defer that refactoring to a secondary PR?

Reply:
Of course, I will mark it with a comment.

Reply:
`res` is `_Py_UCS4` by definition and `_PyUnicode_ExtendedCase` returns a UCS4.